e2e/basic/basic_test.go

```diff
@@ -1,53 +1,62 @@
 package basic
 
 import (
-	"testing"
+	"context"
 
+	"github.com/celestiaorg/knuu/pkg/instance"
 	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/require"
+)
 
-	"github.com/celestiaorg/knuu/pkg/knuu"
+const (
+	TestImage = "alpine:latest"
 )
 
-func TestBasic(t *testing.T) {
-	t.Parallel()
+func (s *Suite) TestBasic() {
+	s.T().Parallel()
 	// Setup
 
-	instance, err := knuu.NewInstance("alpine")
-	if err != nil {
-		t.Fatalf("Error creating instance '%v':", err)
-	}
-	err = instance.SetImage("docker.io/alpine:latest")
-	if err != nil {
-		t.Fatalf("Error setting image: %v", err)
-	}
-	err = instance.SetCommand("sleep", "infinity")
-	if err != nil {
-		t.Fatalf("Error setting command: %v", err)
-	}
-	err = instance.Commit()
-	if err != nil {
-		t.Fatalf("Error committing instance: %v", err)
-	}
+	ctx := context.Background()
+
+	target, err := s.Knuu.NewInstance("alpine")
+	s.Require().NoError(err)
 
-	t.Cleanup(func() {
-		require.NoError(t, knuu.BatchDestroy(instance))
+	s.Require().NoError(target.SetImage(ctx, TestImage))
+	s.Require().NoError(target.SetCommand("sleep", "infinity"))
+	s.Require().NoError(target.Commit())
+
+	s.T().Cleanup(func() {
+		s.T().Log("Tearing down Basic Test...")
+		err := instance.BatchDestroy(ctx, target)
+		if err != nil {
+			s.T().Logf("error destroying instances: %v", err)
+		}
 	})
 
-	// Test logic
+	// Test Logic
+	s.Require().NoError(target.Start(ctx))
+	s.Require().NoError(target.WaitInstanceIsRunning(ctx))
+
+	s.Require().NoError(err)
 
-	err = instance.Start()
-	if err != nil {
-		t.Fatalf("Error starting instance: %v", err)
+	//Perform the test
+	type testCase struct {
+		name string
 	}
-	err = instance.WaitInstanceIsRunning()
-	if err != nil {
-		t.Fatalf("Error waiting for instance to be running: %v", err)
+
+	tt := []testCase{
+		{"Test Case 1"},
+		{"Test Case 2"},
```

Comment:
For this test, we really just need 1 test case, so we can just directly call what is on lines 55 to 58